docs/proposals/api-group.md

```diff
@@ -0,0 +1,104 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<strong>
+The latest 1.0.x release of this document can be found
+[here](http://releases.k8s.io/release-1.0/docs/proposals/api-group.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Supporting multiple API groups
+
+## Goal
+
+1. Breaking the monolithic v1 API to modular groups
+
+2. Supporting different versions in different groups
+
+3. Supporting identically named kinds to exist in different groups
+
+4. Enabling/disabling groups individually
+
+5. Laying the basis for [API Plugin](../../docs/design/extending-api.md)
+
+6. Allowing users to omit group name when using kubectl
+
```

Comment:
Is there a reason that configuration has to be in the same scheme?
Configuration seems very specific to each thing being configured, having
separate schemes there doesn't seem like an issue.

On Thu, Aug 20, 2015 at 1:56 PM, Brian Grant <notifications@github.com>
wrote:

> In docs/proposals/api-group.md
> <https://github.com/kubernetes/kubernetes/pull/12951#discussion_r37560164>
> :
>
> > +# Supporting multiple API groups
> > +
> > +## Goal
> > +
> > +1. Breaking the monolithic v1 API to modular groups
> > +
> > +2. Supporting different versions in different groups
> > +
> > +3. Supporting identically named kinds to exist in different groups
> > +
> > +4. Enabling/disabling groups individually
> > +
> > +5. Laying the basis for [API Plugin](../../docs/design/extending-api.md)
> > +
> > +6. Allowing users to omit group name when using kubectl
> > +
>
> We also want to use groups for component configuration #12245
> <https://github.com/kubernetes/kubernetes/issues/12245>. kubeconfig and
> the scheduler config are existing configuration "APIs" that have no
> explicit groups.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/12951/files#r37560164>.
>



-- 
Clayton Coleman | Lead Engineer, OpenShift